### PR TITLE
Fabo/1517 fix self bond querying

### LINKED ISSUE #1517: Can't undelegate: "invalid Bech32 prefix; expected cosmos, got cosmosvaloper"
UI Version: 03067df8325c49d19fffe9e7e35521f09c534369

Description:

```
$ yarn build:gaia
$ yarn build:local --overwrite=true
$ yarn start local-testnet
```

1. Log in using the `local` account.
1. Undelegate 1 atom.

A notification reports success but the undelegation fails:

screen shot 2018-11-05 at 1 18 19 pm

The HTTP response is `invalid Bech32 prefix; expected cosmos, got cosmosvaloper`.

### PR DESCRIPTION
**Pull request by faboweb**
Closes #1517 

_Description:_

❤️ Thank you!

---

* [ ] Added entries in `CHANGELOG.md` with issue # and GitHub username
* [ ] Reviewed `Files changed` in the github PR explorer

### CODE REVIEW

File: app/src/renderer/components/staking/PageValidator.vue [hunk 1]
```diff
@@ -392,8 +392,12 @@ export default {
     }
   },
   watch: {
-    validator(validator) {
-      this.$store.dispatch(`getSelfBond`, validator)
+    validator: {
+      immediate: true,
+      handler(validator) {
+        if (!validator) return
+        this.$store.dispatch(`getSelfBond`, validator)
```
Comment (fedekunze) [inline app/src/renderer/components/staking/PageValidator.vue]: shouldn't we use `mapActions` instead?

Comment (faboweb) [reply app/src/renderer/components/staking/PageValidator.vue]: We haven't discussed a rule for this yet. Did it anyway to push things along.

Comment (faboweb) [reply app/src/renderer/components/staking/PageValidator.vue]: changed it but then e2e tests failed so I reverted it back to not blow the scope.

Comment (fedekunze) [reply app/src/renderer/components/staking/PageValidator.vue]: ok we can discuss it later

**Pull request merged**